cmd/releasego/update-azure-linux.go

```diff
@@ -276,34 +276,20 @@ func updateSpecFile(assets *buildassets.BuildAssets, changelogDate time.Time, sp
 		oldVersion = goversion.New(matches[2])
 	}
 
-	var oldRelease int
+	var oldRelease string
 	if matches := specFileReleaseRegex.FindStringSubmatch(specFileContent); matches == nil {
 		return "", fmt.Errorf("no Release declaration found in spec content")
 	} else {
-		var err error
-		oldRelease, err = strconv.Atoi(matches[2])
-		if err != nil {
-			return "", fmt.Errorf("failed to parse Release number: %w", err)
-		}
+		oldRelease = matches[2]
 	}
 
-	newVersion := assets.GoVersion().MajorMinorPatch()
-	if strings.Contains(newVersion, "$") {
-		return "", fmt.Errorf("new version %q contains unexpected $", newVersion)
-	}
-	specFileContent = specFileVersionRegex.ReplaceAllString(specFileContent, "${1}"+newVersion)
-
-	// For servicing patches, increment release. Azure Linux may have incremented it manually for a
-	// Azure-Linux-specific fix, so this is independent of Microsoft Go releases. When updating to a
-	// new major/minor version (as semver refers to them), reset release to 1.
-	var newRelease int
-	if assets.GoVersion().MajorMinor() != oldVersion.MajorMinor() {
-		newRelease = 1
-	} else {
-		newRelease = oldRelease + 1
+	newVersion, newRelease, err := updateSpecVersion(assets, oldVersion, oldRelease)
+	if err != nil {
+		return "", err
```

Comment:
I assume the caller will deal with empty string just fine?

Reply:
A detailed doc comment for this func might mention "returns the spec file content or an error". The presence of an error determines whether it's meaningful to use the string for anything. Some functions go into more detail about what exactly is returned when it's useful for the func to promise something more specific, but generally it's simply incorrect to depend on the other values that a func returns when it returns an error.

Various idioms come out of that. If there isn't a reason to return in some other way, typically you either:

* `return (something), (something), ..., nil` or
* `return (zero), (zero), ..., err`

In this case, `""` is just the zero string value.

This is a clever proposal to make that idiom easier to write, but unfortunately it didn't make it:

* https://github.com/golang/go/issues/61372